Allow running serverless benchmarks on dev clusters (#14362)
kyma-project/kyma: tests/serverless-bench/README.md

This a quick guide to simplify running the function controller benchmarks tests on in a development environment. This can be useful in several scenarios. For example, benchmarking a new runtime implementation.

To run the these tests, apply the following steps:

- Start with a fresh Kyma cluster. The following minimum components are requireed:
```yaml
---
defaultNamespace: kyma-system
prerequisites:
  - name: "cluster-essentials"
  - name: "istio"
    namespace: "istio-system"
  - name: "certificates"
    namespace: "istio-system"
components:
  - name: "monitoring"
  - name: "serverless"
```

- Apply the functions you need to benchmark from the test [function fixtures](./fixtures/functions/). If you need to benchmark a new function runtime or configuration, you should create a similar function manifest with the same naming pattern. 

- Wait for the functions to be ready.
- Apply the [local-test](./fixtures/local-test/) manifests. This will deploy the following resources:
    - A MySQL Pod and service to store the benchmarking data
    - A grafana datasource to read the benchmark data from the mysql backend.
    - A grafana dashboard configuration to present the data.

> - Edit the [test job spec](./fixtures/serverless-benchmark-job.yaml) to set `USE_LOCAL_MYSQL` to `true`. You can also use `CUSTOM_FUNCTIONS` to test specific functions, instead of testing the full list.

Reviewer: ```suggestion
- Edit the [test job spec](./fixtures/serverless-benchmark-job.yaml) to set `USE_LOCAL_MYSQL` to `true`. You can also use `CUSTOM_FUNCTIONS` to test specific Functions instead of testing the full list.
```

> - Apply the test job. Each test run will create a data point for each defined function.

Reviewer: ```suggestion
- Apply the test job. Each test run creates a data point for each defined Function.
```

> - Once the first test run is done, you can [access grafana](https://kyma-project.io/docs/kyma/latest/04-operation-guides/security/sec-06-access-expose-kiali-grafana#access-kiali-grafana-and-jaeger) and find the `Serverless Controller Benchmarks` dashboard, and check the data.

Reviewer: ```suggestion
- Once the first test run is done, you can [access Grafana](https://kyma-project.io/docs/kyma/latest/04-operation-guides/security/sec-06-access-expose-kiali-grafana#access-kiali-grafana-and-jaeger) and find the `Serverless Controller Benchmarks` dashboard to check the data.
```
